File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int maxProduct = Integer.MIN_VALUE;
        double prefix = 1.0;
        double suffix = 1.0;
        
        for(int i = 0; i < nums.length; i++) {
            if(prefix == 0) prefix = 1.0;
            if(suffix == 0) suffix = 1.0;
            
            prefix *= nums[i];
            suffix *= nums[nums.length - i - 1];
            
            maxProduct = (int)Math.max(maxProduct, Math.max(prefix, suffix)); 
        }
        
        return maxProduct;
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxProduct = float('-inf')
        prefix = 1
        suffix = 1
        
        for i in range(len(nums)):
            if(prefix == 0):
                prefix = 1
            if(suffix == 0):
                suffix = 1
            prefix *= nums[i]
            suffix *= nums[len(nums) - i - 1]
            maxProduct = max(maxProduct, max(prefix,suffix))
        
        return maxProduct
        